src/scene/renderer/render-pass-cookie-renderer.js

```diff
@@ -74,6 +77,52 @@ class RenderPassCookieRenderer extends RenderPass {
         this._quadRendererCube = null;
     }
 
+    static create(renderTarget, cubeSlotsOffsets) {
+
+        Debug.assert(renderTarget);
+
+        // prepare a single render pass to render all quads to the render target
+        const renderPass = new RenderPassCookieRenderer(renderTarget.device, cubeSlotsOffsets);
+        renderPass.init(renderTarget);
+        renderPass.colorOps.clear = false;
+        renderPass.depthStencilOps.clearDepth = false;
+
+        return renderPass;
+    }
+
+    update(lights) {
+
+        // pick lights we need to update the cookies for
+        const filteredLights = this._filteredLights;
+        this.filter(lights, filteredLights);
+
+        // enabled / disable the pass
+        this.executeEnabled = filteredLights.length > 0;
+    }
+
+    filter(lights, filteredLights) {
```

Comment:
I'm guessing this is much faster than `Array#filter`?

Reply:
this avoids temporary array allocation, so why not